### PR TITLE
LPS-151343[CP 2.0] Add Automate testcase

### PR DESCRIPTION
**Pull request by styller1**
https://issues.liferay.com/browse/LPS-151343

### CODE REVIEW

**Comment by styller1** [comment]
ci:test:sf

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/osb-site-initializer-customer-portal-test/src/testFunctional/macros/CP2NavigationMenu.macro [hunk 1]
```diff
@@ -100,4 +100,20 @@ definition {
...
+
+		Type(
+			locator1 = "CP2NavigationMenu#DATA_FIELD",
+			value1 = "${data}");
+
+		Click.clickAt(
+			index = "${index}",
+			locator1 = "CP2NavigationMenu#FILTER_APPLY");
+	}
+
```
Comment (barrosdennys) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/osb-site-initializer-customer-portal-test/src/testFunctional/macros/CP2NavigationMenu.macro:118]: IS there a way for us to create these locators without relying on indexes?

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/osb-site-initializer-customer-portal-test/src/testFunctional/paths/CP2NavigationMenu.path [hunk 2]
```diff
@@ -95,6 +95,36 @@
...
+<tr>
+	<td>SEARCH_FILTER</td>
+	<td>//div[contains(@class,'bg-neutral-1')]//div[@class="input-group-item"]//input[@placeholder="Search"]</td>
+	<td></td>
+</tr>
+<tr>
+	<td>SEARCH_RESULT</td>
+	<td>//p[contains(@class, 'font-weight-semi-bold m-0 mt-3 text-paragraph-sm') and contains(.,'result')]</td>
+	<td></td>
+</tr>
```
Comment (barrosdennys) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/osb-site-initializer-customer-portal-test/src/testFunctional/paths/CP2NavigationMenu.path:107]: @styller1 the way this locator is written, with this class inside the contains, it is not error prone. The likelihood of it to change is high. Change "font-weight-semi-bold m-0 mt-3 text-paragraph-sm" to only "text-paragraph-sm" and check if it works

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/osb-site-initializer-customer-portal-test/src/testFunctional/tests/CP2PortalActivation.testcase [hunk 3]
```diff
@@ -0,0 +1,403 @@
...
+		task ("Signs in OKTA to update and go to Customer Portal") {
+			CustomerPortal.loginOktaToUpdateCP();
+
+			Navigator.openURL();
+
+			ApplicationsMenu.gotoSite(key_site = "${testSiteName}");
+		}
+
+		task ("Select a project and go to activation page") {
+			CustomerPortal.enterOnProject(title = "NASA ICE");
```
Comment (barrosdennys) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/osb-site-initializer-customer-portal-test/src/testFunctional/tests/CP2PortalActivation.testcase:375]: @styller1 Why this test doesnt need the  macro call 

CustomerPortal.search(searchItem = "NASA ICE");

before CustomerPortal.enterOnProject(title = "NASA ICE");

Is it different?

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/osb-site-initializer-customer-portal-test/src/testFunctional/tests/CP2PortalActivation.testcase [hunk 4]
```diff
@@ -0,0 +1,403 @@
...
+
+			Type(
+				locator1 = "CP2NavigationMenu#SEARCH_FILTER",
+				value1 = "192.168.0.111");
+
+			KeyPress(
+				locator1 = "CP2NavigationMenu#SEARCH_FILTER",
+				value1 = "\ENTER");
+
+			AssertElementPresent(locator1 = "CP2NavigationMenu#SEARCH_RESULT");
```
Comment (barrosdennys) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/osb-site-initializer-customer-portal-test/src/testFunctional/tests/CP2PortalActivation.testcase:393]: What are we trying to assert here? If the result of the search is 192.168.0.111? With the current locator of SEARCH_RESULT, it doesn't seem to validate it correctly. IT is not looking for any specific ipAddress as far as I understood.

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/osb-site-initializer-customer-portal-test/src/testFunctional/tests/CP2PortalActivation.testcase [hunk 5]
```diff
@@ -0,0 +1,403 @@
...
+		task ("Search for a IP Addresses and assert the search") {
+			Click(locator1 = "CP2NavigationMenu#SEARCH_FILTER");
+
+			Type(
+				locator1 = "CP2NavigationMenu#SEARCH_FILTER",
+				value1 = "192.168.0.111");
+
+			KeyPress(
+				locator1 = "CP2NavigationMenu#SEARCH_FILTER",
+				value1 = "\ENTER");
```
Comment (barrosdennys) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/osb-site-initializer-customer-portal-test/src/testFunctional/tests/CP2PortalActivation.testcase:391]: Can we create a macro for it with the ip as parameter?

**Comment by styller1** [comment]
ci:test:sf

**Comment by styller1** [comment]
ci:test:sf

**Comment by styller1** [comment]
ci:test:sf

**Comment by styller1** [comment]
ci:test:sf

**Comment by styller1** [comment]
ci:test:sf

**Comment by kevenleone** [comment]
Archiving due the inactivity

**Pull request closed without merging**